linkatos/confirmation.py

```diff
@@ -9,12 +9,19 @@ def update_if_url(parsed_message, expecting_confirmation):
     return (expecting_confirmation)
 
 
-def process_if_yn(parsed_message, expecting_confirmation):
-    if expecting_confirmation is False or parsed_message['type'] is not 'yn_answer':
-        is_yes = False
-        return (expecting_confirmation, is_yes)
+def evaluate(expecting_confirmation, parsed_message, item_ts):
+    # returns (@expecting_confirmation: boolean, @confirmation: boolean)
+    if expecting_confirmation is False or \
+       parsed_message['type'] is not 'reaction' or \
+       parsed_message['item_ts'] != item_ts:
+        return (expecting_confirmation, False)
 
-    is_yes = parsed_message['out']  # True if 'yes', False if 'no'
-    expecting_confirmation = False
+    if parsed_message['message'] == '+1':
+        return (False, True)
 
-    return (expecting_confirmation, is_yes)
+    if parsed_message['message'] == '-1':
+        return (False, False)
+
+    # when we're expecting a confirmation and it's not a thumbs up or a thumbs
+    # down, we keep waiting
+    return (expecting_confirmation, False)
```

Comment:
Trying to follow this idea I've made some changes.
Basically took the `expecting_confirmation` out of this function.
The bit which I'm still not comfortable with is the fact that there is a need of updating the `expecting_confirmation` depending on whether or not there was an actual reaction, and as long as this is evaluated inside the `evaluate` function I need to differentiate that outcome from a `False`. 
I've used `None` but, I'm not sure if I should separate all the conditions that determine whether or not the message is a reaction.